disnake/invite.py

```diff
@@ -401,6 +410,19 @@ def __init__(
             data.get("channel"), channel
         )
 
+        if (
+            self.guild is not None
+            and (guild_data := data.get("guild"))
+            and "welcome_screen" in guild_data
+        ):
+            self.guild_welcome_screen: Optional[WelcomeScreen] = WelcomeScreen(
+                state=self._state,
+                data=guild_data["welcome_screen"],
+                guild=self.guild,  # type: ignore
```

Comment:
Supporting `PartialInviteGuild` in `WelcomeScreen` might be preferable, since it's a private attr anyway

Reply:
typehint hadn't been removed https://github.com/DisnakeDev/disnake/pull/339/commits/5cf49c757bee4b0737780a22d5d852425549708f

Reply:
It is supported, but there because `self.guild` can be a `disnake.Object`